.github/workflows/unit-test.yml

```diff
@@ -0,0 +1,33 @@
+name: Run and Validate Unit Tests
+
+on:
+  push:
+    branches: 
+      - '**'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Execute Unit Tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 1
+      matrix:
+        python-version:
+          - 3.8
+        os:
+          - ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@master
```

Comment:
Looks like testing is only happening on the master branch, regardless of the current commit branch